### ISSUE TITLE
AsMiddleware will ignore the abstract definition's tags

### ISSUE THREAD

**Issue by ywisax** [doctrine/DoctrineBundle#1817]
Sample code:

```
<?php

namespace App\Doctrine\Middleware;

use App\Doctrine\Service\QueryExecutionTimeLogger;
use Doctrine\Bundle\DoctrineBundle\Attribute\AsMiddleware;
use Doctrine\DBAL\Driver;
use Doctrine\DBAL\Driver\Middleware;
use Symfony\Component\Stopwatch\Stopwatch;
use Symfony\Contracts\Service\ResetInterface;

#[AsMiddleware]
class LogMiddleware implements Middleware, ResetInterface
{
    private Stopwatch $stopwatch;

    public function __construct(
        private readonly QueryExecutionTimeLogger $timeLogger,
    ) {
        $this->stopwatch = new Stopwatch(true);
    }

    public function wrap(Driver $driver): Driver
    {
        return new LogDriver($driver, $this->timeLogger, $this->stopwatch);
    }

    public function reset(): void
    {
        var_dump('LogMiddleware reset');
        debug_print_backtrace();
        $this->stopwatch->reset();
    }
}
```

We have three connection profile, so the \Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\MiddlewaresPass will create three new service, like:

![image](https://github.com/user-attachments/assets/b621688a-bb1d-46fd-90db-aaa3bc02988d)

And the abstract middleware service's reset method will not trigger, because there are no kernel.reset tag like this:

![image](https://github.com/user-attachments/assets/4ba0d0f7-ea09-4360-90c4-a19ac628ef9a)

**Comment by ostrolucky** [comment]
I've a fix for this at https://github.com/doctrine/DoctrineBundle/pull/1819. Would you try it if that solves your issues?

**Comment by ywisax** [comment]
It works!
![image](https://github.com/user-attachments/assets/fa1627d6-b728-48fe-9aa4-5b8e2e3f9857)


> I've a fix for this at #1819. Would you try it if that solves your issues?
